### ISSUE TITLE
About the release semantics

### ISSUE THREAD

**Issue by chenrui333** [c3lang/c3c#1143]
👋 We are trying to ship c3c 0.5.4 on the brew side, but there is some concern that it is a pre-release, per what I saw, the 0.5 release was also marked as `pre release`, want to raise this issue to check if that is intentional or not. Thanks!

relates to https://github.com/Homebrew/homebrew-core/pull/163000

**Comment by lerno** [comment]
It's a good question. According to github "pre-release" is for something not production ready. Which would be anything before 1.0. On the other hand, after release, it's useful to give a few days of grace period before even something like 0.5.4 goes golden. I don't know if pre-release is good for that or if I just should use 'latest' for that.

Actually I did just patch 0.5.4 due to standard library changes that just had to go in (regressions).

In any case I'd like to be able to support the work for homebrew well. Do you have any suggestions?

**Comment by chenrui333** [comment]
> According to github "pre-release" is for something not production ready.

if you do semver, then anything before 1.0.0 would be pre-release. I actually feel like folks using pre-release. In that sense, I would think you can just use latest release, and pre-release for any real pre-release testing. :) 

Let me know if that helps.

**Comment by lerno** [comment]
Uh, no I'm not quite sure what you recommend could you rephrase it?

**Comment by SMillerDev** [comment]
> Actually I did just patch 0.5.4 due to standard library changes that just had to go in (regressions).

Since releases are just GitHub tags, that would generally be treated as a 0.5.4.1 since
the [git manual](https://git-scm.com/docs/git-tag#_on_re_tagging) says re-tagging is "the insane thing" to do.

---

Generally projects take the following approach:
- release: as stable as it gets for the project
- pre-release: use only if you know what you are doing, don't expect any support if you find bugs. A release of this will come later. (think alpha/beta/RC)

If that is also the approach you take maybe Homebrew should not be shipping the pre-releases yet. If you consider the pre-releases stable, it might be easiest to just let the 0.5.x semver speak to the stability of the project.

**Comment by lerno** [comment]
@SMillerDev just to be clear, I wasn't aware of Brew being updated, which is why I allowed myself to re-tag. Now I know better.

It seem that maybe the best thing is to do like this:
1. Initial merge of 0.5.4 goes to "latest" and is built as such.
2. Testing of the binary for people not compiling dev (2-3 days)
3. Actual tag this "latest" as the 0.5.4 release and tag it as a release (and not pre-release)

**Comment by SMillerDev** [comment]
The clarification is much appreciated!

**Comment by lerno** [comment]
@SMillerDev would the above suggested procedure work well?

**Comment by SMillerDev** [comment]
Yeah, sounds good. As long as releases are static and marked as release you've certainly made life easier for brew. Thanks a lot!

**Comment by goyalyashpal** [comment]
so, close as completed?

**Comment by chenrui333** [comment]
Yes, all good. Thanks folks!! 🙏

**Comment by lerno** [comment]
@chenrui333 I noticed that for 0.6.0 the binaries for macos are aarch64. I'm working to change this ASAP

**Comment by SMillerDev** [comment]
Homebrew compiles its own binaries so we're not affected. Thanks for the heads up though!
